include/dmlc/concurrency.h

```diff
@@ -28,7 +28,7 @@ class Spinlock {
     lock_.clear();
   }
 #else
-  Spinlock() : lock_(ATOMIC_FLAG_INIT) {
+  Spinlock() : lock_() {
```

Comment:
I think better to have in the class declaration:
std::atomic_flag lock_ = ATOMIC_FLAG_INIT;
Doesn't make sense to have an empty initializer at the constructor level like that,m since the effect is the same and we're making assumptions about the value of ATOMIC_FLAG_INIT

Reply:
Can't use = ../dmlc-core/include/dmlc/concurrency.h:32:11: error: use of deleted function ‘std::atomic_flag& std::atomic_flag::operator=(const std::atomic_flag&)’

Reply:
I fixed the warning in mxnet by disabling that specific warning as there's no good way to remove the warning.

Reply:
lock_(ATOMIC_FLAG_INIT[0]) ?
Or you can just assign it within the constructor code block.

Reply:
I think just call clear() in the constructor and be done with it.